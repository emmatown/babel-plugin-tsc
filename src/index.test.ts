import { transformFileAsync } from "@babel/core";
import path from "path";
import * as fs from "fs-extra";

let babelOptions = {
  configFile: false,
  babelrc: false,
  plugins: [require("./index")]
} as any;

test("it works", async () => {
  let result = await transformFileAsync(
    path.resolve(__dirname, "..", "test", "one.ts"),
    babelOptions
  );
  expect(result!.code).toMatchSnapshot();
});

test("it works if things change", async () => {
  let origTest = path.resolve(__dirname, "..", "test");
  let testCopy = path.resolve(__dirname, "..", "test-copy");
  await fs.remove(testCopy);
  await fs.copy(origTest, testCopy);
  let fileAToTransform = path.resolve(testCopy, "one.ts");
  let result = await transformFileAsync(
    path.resolve(testCopy, "one.ts"),
    babelOptions
  );
  expect(result!.code).toMatchSnapshot();
  await fs.writeFile(
    fileAToTransform,
    `import { OtherThing } from "./another";

  const enum Thing {
    one,
    two,
    three
  }
  
  let thing = Thing.three;
  
  let otherThing = OtherThing.oneMore;
  `
  );
  await fs.writeFile(
    path.resolve(testCopy, "another.ts"),
    `export const enum OtherThing {
      another,
      thing,
      oneMore
    }
    `
  );
  let secondResult = await transformFileAsync(
    path.resolve(testCopy, "one.ts"),
    babelOptions
  );
  expect(secondResult!.code).toMatchSnapshot();
});
