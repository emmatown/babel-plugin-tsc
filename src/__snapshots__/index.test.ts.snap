// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it works 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
let thing = 0
/* one */
;
let otherThing = 1
/* thing */
;"
`;

exports[`it works if things change 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
let thing = 0
/* one */
;
let otherThing = 1
/* thing */
;"
`;

exports[`it works if things change 2`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
let thing = 2
/* three */
;
let otherThing = 2
/* oneMore */
;"
`;
